R/alexanderZTest.R

```diff
@@ -0,0 +1,166 @@
+    stop("Paired and two-sample z.test not yet implemented")
+  
+  result <- list(statistic=NULL, p.value=NULL, conf.int=NULL, estimate=NULL, null.value=mu, 
+                 alternative=alternative, method=NULL, data.name=NULL, 
+                 d=NULL, confIntEffSize=NULL)
+  
+  nX <- length(x)
+  nY <- NULL
+  
+  if (nX <= 1) 
+    stop("not enough x observations")
+  
+  xBar <- mean(x)
+  
+  if (is.null(y)) {
+    dname <- deparse(substitute(x))
+    
+    method <- "One-sample z-Test"
+    estimate <- xBar - mu
+    names(estimate) <- "mean difference of x"
+    d <- (xBar-mu)/sigma.x
+    zStat <- sqrt(nX) * d
+    sdError <- sigma.x/sqrt(nX)
+  } else {
+    dname <- paste(deparse(substitute(x)), "and", paste(deparse(substitute(y))))
+    
+    if (paired==TRUE) {
+      stop("Not yet implemented")
+      method <- "Paired z-test"
+    } 
+    
+    if (paired==FALSE) {
+      nY <- length(y)
+      
+      if (nY <= 1) 
+        stop("not enough y observations")
+      
+      yBar <- mean(y)
+      method <- "Two-sample z-Test"
+      estimate <- c(xBar, yBar)-mu
+      names(estimate) <- c("mean difference of x", "mean difference of y")
+      
+      sdError <- sqrt(((sigma.x^2)/nX) + ((sigma.y^2)/nY))
+      zStat <- (xBar - yBar - mu)/stderr
+      nEff <- (1/nX + 1/nY)^(-1)
+      d <- zStat/sqrt(nEff)
+    }
+  }
+  
+  # TODO(Alexander): Doesn't work for two-sample problems. Furthermore, only scaled wrt sigma.x
+  # 
+  # args(.zMeanDifferenceConfidenceIntervalStat)
+  
+  
+  tempResult <- .zMeanDifferenceConfidenceIntervalStat("zStat"=zStat, "ciValueMeanDiff"=conf.level,
+                                                       "sdError"=sdError, "nX"=nX, "nY"=nY, "alternative"=alternative,
+                                                       "ciValueMeanDiffES"=ciValueMeanDiffES)
+  confInt <- tempResult[["confInt"]]
+  confIntEffSize <- tempResult[["confIntEffSize"]]
+  
+  pValue <- switch(alternative,
+                   "two.sided"=2 * pnorm(-abs(zStat)),
+                   "less"=pnorm(zStat),
+                   "greater"=1-pnorm(zStat))
+  
+  result[["statistic"]] <- zStat
+  result[["p.value"]] <- pValue
+  result[["conf.int"]] <- confInt
+  result[["estimate"]] <- estimate
+  result[["d"]] <- d
+  result[["confIntEffSize"]] <- confIntEffSize
+  result[["method"]] <- method
+  result[["data.name"]] <- dname
+  return(result)
+}
+
+
+.zMeanDifferenceNonScaledConfidenceInterval <- function(zStat, alternative, ciValue) {
+  result <- c(NA, NA)
+  tempAlpha <- if (alternative == "two.sided") {
+    (1-ciValue)/2
+  } else {
+    1-ciValue
+  }
+  
+  deviation <- qnorm(1-tempAlpha)
+  
+  if (alternative=="two.sided") {
+    result[1] <- zStat - deviation
+    result[2] <- zStat + deviation
+  } else if (alternative=="greater") {
+    result[1] <- zStat - deviation
+    result[2] <- Inf
+  } else if (alternative=="less") {
+    result[1] <- -Inf
+    result[2] <- zStat + deviation
+  }
+  return(result)
+}
```

Comment:
shouldn't this also use the given population sd? For example, 
```r
if (alternative=="two.sided") {
  result[1] <- zStat - sd * deviation
  result[2] <- zStat + sd * deviation
}
```
The reason I'm wondering is that I noticed that the confidence intervals of the effect size for the z-test don't really change when you adjust the 95% ci in the options (even though it changes a bit for the Student t-test).

Reply:
The reason why it doesn't change is because the standard confidence interval for the mean difference here depends on conf.int argument (R style, see t.test), whereas the confidence interval for the mean difference effect size depends on ciValueMeanDiffES. 

The use case for different ciValues is because of our UI. To simplify matters, I:
1. For readability purposes I removed `.zMeanDifferenceNonScaledConfidenceInterval`. This is less efficient, but we don't live in the 1950s anymore. 
2. Unified ciValue names. Now ciValueMeanDiff and ciValueESMeanDiff (no more conf.int R style)
3. Default call `ciValueESMeanDiff=ciValueMeanDiff`, which is helpful in R, but doesn't matter for JASP
4. Scaling now aligned with the D statistic. 

```
set.seed(1)
x <- rnorm(10)

bob <- .z.test("x"=x)
bob$conf.int
bob$confIntEffSize

bob <- .z.test("x"=x, ciValueMeanDiff=0.6)
bob$conf.int
bob$confIntEffSize

bob <- .z.test("x"=x, ciValueMeanDiff=0.1, ciValueESMeanDiff=0.6)
bob$conf.int
bob$confIntEffSize
```